### PR TITLE
[Kotlin Migration] BadgeDrawableBuilder

### PR DESCRIPTION
**Pull request by puranjayK**
[Kotlin Migration] BadgeDrawableBuilder
package com.ichi2.ui

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/ui/BadgeDrawableBuilder.kt [hunk 1]
```diff
@@ -14,86 +14,66 @@
...
+import android.content.res.Resources
+import android.graphics.drawable.Drawable
+import android.os.Build
+import android.view.MenuItem
+import androidx.vectordrawable.graphics.drawable.VectorDrawableCompat
+import com.ichi2.anki.R
+import timber.log.Timber
+
+class BadgeDrawableBuilder(private val mResources: Resources) {
+    private var mChar = 0.toChar()
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/ui/BadgeDrawableBuilder.kt:28]: unsure. '\u0000' at the very least, but needs investigation

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
```
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/ui/BadgeDrawableBuilder.kt:27: Error: Variable name should not use field prefixes. [VariableNamingDetector from com.ichi2.anki.lint]
class BadgeDrawableBuilder(private val mResources: Resources) {
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: AnkiDroid/src/main/java/com/ichi2/ui/BadgeDrawableBuilder.kt (hunk 1 above)
Comment (puranjayK) [reply AnkiDroid/src/main/java/com/ichi2/ui/BadgeDrawableBuilder.kt:28]: The java code was `mChar !='\0'`. 
I got an [answer ](https://stackoverflow.com/questions/36269488/how-do-i-make-a-null-character-in-kotlin)on stackoverflow regarding the same. 
Even from kotlinlang, I got to understand that `'\u0000'` is null char

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/ui/BadgeDrawableBuilder.kt:28]: `\u0000` looks better to me: https://stackoverflow.com/questions/36269488/how-do-i-make-a-null-character-in-kotlin

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by Arthur-Milchior — APPROVED** [review]

**Pull request merged**